### PR TITLE
add support for Lucee 6.1 and use Java 21 for Lucee 6

### PR DESCRIPTION
**Pull request by michaeloffner**

### CODE REVIEW

**Comment by zspitzer** [comment]
Yes, but Java 21 isn't supported yet, date handling is broken by Unicode changes https://luceeserver.atlassian.net/browse/LDEV-4658

**Comment by michaeloffner** [comment]
@zspitzer that is expected and not a bug, date/time format always change with every java version, that is expected. solution is to define java versions in test cases.
This has nothing to do with Lucee, that is simply the difference in the implementation of "SimpleDateFormat" between versions.

**Comment by michaeloffner** [comment]
@zspitzer also that is nothing that should stop this, 21 is a requirement for Lucee 6, no way around it.

**Comment by michaeloffner** [comment]
@zspitzer that ticket is resolved now

File: matrix.yaml [hunk 1]
```diff
@@ -139,3 +160,15 @@ exclusions:
...
   - TOMCAT_JAVA_VERSION: 'jre8-temurin-jammy'
     LUCEE_MINOR: '6.0'
+  - TOMCAT_JAVA_VERSION: 'jdk8-temurin-jammy'
+    LUCEE_MINOR: '6.0'
+  - TOMCAT_JAVA_VERSION: 'jre8-temurin-jammy'
+    LUCEE_MINOR: '6.1'
+  - TOMCAT_JAVA_VERSION: 'jdk8-temurin-jammy'
+    LUCEE_MINOR: '6.1'
+  - TOMCAT_JAVA_VERSION: 'jre21-temurin-jammy'
+    LUCEE_MINOR: '5.3'
```
Comment (zspitzer) [inline matrix.yaml:172]: why?

File: matrix.yaml [hunk 2]
```diff
@@ -139,3 +160,15 @@ exclusions:
...
+  - TOMCAT_JAVA_VERSION: 'jre21-temurin-jammy'
+    LUCEE_MINOR: '5.4'
```
Comment (zspitzer) [inline matrix.yaml:177]: not supported

File: matrix.yaml [hunk 3]
```diff
@@ -139,3 +160,15 @@ exclusions:
   # Java 8 is not needed for Lucee 6.0
   - TOMCAT_JAVA_VERSION: 'jre8-temurin-jammy'
     LUCEE_MINOR: '6.0'
+  - TOMCAT_JAVA_VERSION: 'jdk8-temurin-jammy'
+    LUCEE_MINOR: '6.0'
+  - TOMCAT_JAVA_VERSION: 'jre8-temurin-jammy'
+    LUCEE_MINOR: '6.1'
```
Comment (zspitzer) [inline matrix.yaml:167]: nope, 6.1 only java 11 and 21 i'd say?

**Review by zspitzer — CHANGES_REQUESTED** [review]

File: matrix.yaml (hunk 1 above)
Comment (michaeloffner) [reply matrix.yaml:172]: @zspitzer we do not Java 8 build for Lucee 6 and no java 21 builds for Lucee 5

**Comment by michaeloffner** [comment]
@zspitzer there was indeed a mistake, i have fixed it, it now looks like this
```
exclusions:

  # Tomcat 10.1 is not needed for Lucee 5.3
  # - TOMCAT_VERSION: '10.1'
  #   LUCEE_MINOR: '5.3'

  # Zero variant is not needed for Lucee 5.3/5.4
  # - LUCEE_MINOR: '5.3'
  #   LUCEE_VARIANT: '-zero'
  # - LUCEE_MINOR: '5.4'
  #   LUCEE_VARIANT: '-zero'

  # Java 8 is not needed for Lucee 6.0
  - TOMCAT_JAVA_VERSION: 'jre8-temurin-jammy'
    LUCEE_MINOR: '6.0'
  - TOMCAT_JAVA_VERSION: 'jdk8-temurin-jammy'
    LUCEE_MINOR: '6.0'
  # Java 8 is not needed for Lucee 6.1
  - TOMCAT_JAVA_VERSION: 'jre8-temurin-jammy'
    LUCEE_MINOR: '6.1'
  - TOMCAT_JAVA_VERSION: 'jdk8-temurin-jammy'
    LUCEE_MINOR: '6.1'
  # Java 21 is not needed for Lucee 5.3
  - TOMCAT_JAVA_VERSION: 'jre21-temurin-jammy'
    LUCEE_MINOR: '5.3'
  - TOMCAT_JAVA_VERSION: 'jdk21-temurin-jammy'
    LUCEE_MINOR: '5.3'
  # Java 21 is not needed for Lucee 5.4
  - TOMCAT_JAVA_VERSION: 'jre21-temurin-jammy'
    LUCEE_MINOR: '5.4'
  - TOMCAT_JAVA_VERSION: 'jdk21-temurin-jammy'
    LUCEE_MINOR: '5.4'
```

so excluding Java 21 for Lucee 5 and excluding Java 8 for Lucee 6

**Pull request closed without merging**